docs/api-aggregation-setup.md

```diff
@@ -0,0 +1,196 @@
+running as a separate pod in kubernetes v1.6. The aggregator is
+enabled by default in kubernetes v1.7 as a beta API, and is the
+default secure endpoint directly integrated into the core kubernetes
+APIServer.
+
+For background on why we're messing with certificates and CA's at all,
+please check [the auth doc](auth.md).
+
+# Steps
+
+## Prerequisites
+You need `cfssl` tools, install with `go get -u github.com/cloudflare/cfssl/cmd/...`.
+
+## Check that aggregator is enabled
+
+`kubectl api-versions` MUST list `apiregistration.k8s.io/v1beta1` (or
+`apiregistration.k8s.io/v1alpha1` if running in kubernetes v1.6)
+
+This won't show up if you are talking to the insecure port. The
+insecure port is not behind the aggregator. The aggregator does not
+support routing of requests to the insecure port. You must talk to the
+secure port, as the aggregator does not serve an API on the insecure
+port.
+
+## Variables used during the rest of the steps
+
+These will be used during the steps and for the final `helm
+install`. They are key to set, as the signed certificates will be
+bound to the exact service name that is defined below as
+`SVCCAT_SERVICE_NAME`.
+
+```
+export HELM_NAME=catalog
+export SVCCAT_NAMESPACE=catalog
+export SVCCAT_SERVICE_NAME=${HELM_NAME}-${SVCCAT_NAMESPACE}-apiserver
+```
+ 
+## Set up the certificate bundle
+
+The APIService expects a certificate bundle. We can create our own, or
+pull the one from kube core for reuse.
+
+The certificate bundle is made up of Certificate Authority, a Serving
+Certificate, and the Serving Private Key.
+
+### Create our own new CA and generate keys
+
+This is an example. It is written with zero understanding of the best
+practices of secure certificate generation.
+
+```
+export CA_NAME=ca
+
+export ALT_NAMES="\"${SVCCAT_SERVICE_NAME}.${SVCCAT_NAMESPACE}\",\"${SVCCAT_SERVICE_NAME}.${SVCCAT_NAMESPACE}.svc"\"
+
+export SVCCAT_CA_SETUP=svc-cat-ca.json
+cat >> ${SVCCAT_CA_SETUP} << EOF
+{
+    "hosts": [ ${ALT_NAMES} ],
+    "key": {
+        "algo": "rsa",
+        "size": 2048
+    },
+    "names": [
+        {
+            "C": "US",
+            "L": "san jose",
+            "O": "kube",
+            "OU": "WWW",
+            "ST": "California"
+        }
+    ]
+}
+EOF
+
+
+cfssl genkey --initca ${SVCCAT_CA_SETUP} | cfssljson -bare ${CA_NAME}
+# now the files 'ca.csr  ca-key.pem  ca.pem' exist
+
+export SVCCAT_CA_CERT=${CA_NAME}.pem
+export SVCCAT_CA_KEY=${CA_NAME}-key.pem
+
+export PURPOSE=server
+echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+
+echo '{"CN":"'${SVCCAT_SERVICE_NAME}'","hosts":['${ALT_NAMES}'],"key":{"algo":"rsa","size":2048}}' \
+ | cfssl gencert -ca=${SVCCAT_CA_CERT} -ca-key=${SVCCAT_CA_KEY} -config=server-ca-config.json - \
+ | cfssljson -bare apiserver
+
+export SC_SERVING_CA=${SVCCAT_CA_CERT}
+export SC_SERVING_CERT=apiserver.pem
+export SC_SERVING_KEY=apiserver-key.pem
+```
```

Comment:
Nit: I think if you go this route the APIService resource needs to set `InsecureSkipTLSVerify` since the CA will necessarily not be in the aggregator's trust store.